webapp/channels/src/components/new_search/select_team.tsx

```diff
@@ -0,0 +1,149 @@
+    const isDifferentTeamSelected = props.selectedTeamId !== currentTeamId;
+
+    const allTeamsText = intl.formatMessage({
+        id: 'search_teams_selector.all_teams',
+        defaultMessage: 'All Teams',
+    });
+
+    let menuButtonText = allTeamsText;
+    if (props.selectedTeamId) {
+        const team = myTeams.find((t) => t.id === props.selectedTeamId);
+        if (team) {
+            menuButtonText = team.display_name;
+        }
+    }
+
+    const button = (
+        <><span>{menuButtonText}</span> <ChevronDownIcon size={12}/></>
+    );
+
+    const onFilterChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setFilter(e.target.value);
+    };
+
+    const handleTeamChange = (teamId: string) => {
+        props.onTeamSelected(teamId);
+        setFilter('');
+    };
+
+    // we only show the filter input and separator if there's more than 4 teams
+    const showFilter = myTeams.length > 4;
+
+    const teams = React.useMemo(() => {
+        // if we show the filter, exclude the current team from the list
+        const myTeamsList = showFilter ? myTeams.filter((team) => team.id !== props.selectedTeamId) : myTeams;
+
+        let filteredTeams = filterAndSortTeamsByDisplayName(myTeamsList, locale, userTeamsOrderPreference);
+        if (filter) {
+            filteredTeams = filteredTeams.filter((team) => team.display_name.toLowerCase().includes(filter.toLowerCase()));
+        }
+        return filteredTeams;
+    }, [myTeams, locale, userTeamsOrderPreference, filter, showFilter, props.selectedTeamId]);
+
+    const renderTeam = (teamId: string, teamName: string, elementId: string, className: string = '') => {
+        return (
+            <Menu.Item
+                id={elementId}
+                role='menuitemradio'
+                forceCloseOnSelect={true}
+                aria-checked={teamId === props.selectedTeamId}
+                key={'team-' + teamId}
+                onClick={() => handleTeamChange(teamId)}
+                labels={<span>{teamName}</span>}
+                trailingElements={(teamId === props.selectedTeamId && (
+                    <CheckIcon
+                        size={14}
+                        color='var(--button-bg, #1c58d9)'
+                    />
+                ))}
+                className={className}
+            />
+        );
+    };
+
+    // MUI Menu doesn't support fragments, and the recommended alternative is to use an array.
+    const renderFilterArea = () => {
+        const elements = [
+            <Menu.Input
+                key='filter_teams'
+                id='search_teams'
+                type='text'
+                placeholder={intl.formatMessage({id: 'search_teams_selector.search_teams', defaultMessage: 'Search teams'})}
+                className='search-teams-selector-search'
+                inputPrefix={<SearchIcon size={12}/>}
+                value={filter}
+                onChange={onFilterChange}
+            />,
```

Comment:
While this doesn't appear to break the a11y of the team selector menu per se, this input isn't currently accessible.

https://github.com/user-attachments/assets/883462ea-bc6e-4676-b961-ed51175a6dfe

Reply:
I created a ticket for this and will deal in a followup PR: https://mattermost.atlassian.net/browse/MM-62682